### ISSUE TITLE
Incorrect method type argument inference

### ISSUE THREAD

**Issue by smillst** [typetools/checker-framework#531]
For this code:

```
import java.util.*;
import java.util.stream.*;

public class Test {

    public List<String> test(List<String> list) {
        return list.stream().collect(Collectors.toList());
    }
}
```

I get NullnessChecker error:

```
Test.java:7: error: [return.type.incompatible] incompatible types in return.
        return list.stream().collect(Collectors.toList());
                                    ^
  found   : @Initialized @NonNull List<? extends @Initialized @Nullable Object>
  required: @Initialized @NonNull List<@Initialized @NonNull String>
1 error
```

(Issue submitted by @trask)

**Comment by mernst** [comment]
Until this is fixed, a workaround is to explicitly write the type argument.

**Comment by smillst** [comment]
This code now crashes.  See #953

**Comment by smillst** [comment]
This false positives should go away once #979 is fixed. (These test cases will be added to that issue.)
